src/core/core/rpc_engine.cpp

```diff
@@ -846,14 +850,28 @@ namespace dsn {
                             {
                                 resolver->on_access_failure(req2->header->gpid.u.partition_index, err);
 
+                                /*
+                                // TODO(qinzuoyan): because the user will fetch reply states from this response_task 'call',
```

Comment:
but wait_and_unwrap() waits the first created response_task:
```
    std::pair< ::dsn::error_code, update_response> put_sync(
        const update_request& args,
        std::chrono::milliseconds timeout = std::chrono::milliseconds(0), 
        uint64_t hash = 0,
        dsn::optional< ::dsn::rpc_address> server_addr = dsn::none
        )
    {
        return ::dsn::rpc::wait_and_unwrap<update_response>(
            ::dsn::rpc::call(
                server_addr.unwrap_or(_server),
                RPC_RRDB_RRDB_PUT,
                args,
                nullptr,
                empty_callback,
                hash,
                timeout,
                0
                )
            );
    }
```
and the retry created response_task will not be handled.